ttnn/cpp/ttnn/operations/uniform/device/uniform_program_factory.cpp

```diff
@@ -0,0 +1,153 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+#include "common/constants.hpp"
+#include "common/core_coord.h"
+#include "common/tt_backend_api_types.hpp"
+#include "tt_metal/common/work_split.hpp"
+#include "ttnn/tensor/types.hpp"
+#include "uniform_device_operation.hpp"
+
+namespace ttnn::operations::uniform {
+
+using namespace tt;
+using namespace tt::tt_metal;
+
+uint32_t get_random_seed() {
+    std::mt19937 rng(std::time(0));
+    std::uniform_int_distribution d(1, 1 << 20);
+    return d(rng);
+}
+
+UniformDeviceOperation::Factory::cached_program_t UniformDeviceOperation::Factory::create(
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& output) {
+    Device* device = output.device();
+    auto grid = device->compute_with_storage_grid_size();
+    int core_h = grid.y;
+
+    uint32_t units_to_divide = output.volume() / constants::TILE_HEIGHT / constants::TILE_WIDTH;
+    auto [num_cores, all_cores, core_group_1, core_group_2, units_per_core_group_1, units_per_core_group_2] =
+        split_work_to_cores(grid, units_to_divide);
+
+    Program program = Program();
+
+    DataType output_dtype = output.dtype();
+    tt::DataFormat out_data_format = datatype_to_dataformat_converter(output_dtype);
```

Comment:
```suggestion
    DataFormat out_data_format = datatype_to_dataformat_converter(output_dtype);
```

Reply:
<img width="587" alt="image" src="https://github.com/user-attachments/assets/a44dddb9-ec30-4031-9be5-90b7a66b2cab">
There are 2 candidates for DataFormat. I have to use tt:: to specify which one to use.

Reply:
Wow, this is another issue. For now, I think we can change it to "auto."

Reply:
Updated